test/ChirpDatabase.Tests/ChirpDatabaseContextTest.cs

```diff
@@ -0,0 +1,184 @@
+// Code inspired by https://blog.jetbrains.com/dotnet/2023/10/24/how-to-use-testcontainers-with-dotnet-unit-tests/
+using Chirp.Infrastructure;
+using Microsoft.EntityFrameworkCore;
+using Chirp.Infrastructure.Models;
+using Chirp.Web;
+
+namespace ChirpDatabase.Tests;
+
+public class ChripDatabaseContextTest : IClassFixture<DatabaseFixture>
+{
+    private readonly ChirpDBContext context;
+
+    public ChripDatabaseContextTest(DatabaseFixture _fixture)
+    {
+        context = _fixture.GetContext();
+    }
+
+    // Users tests
+    [Theory]
+    [InlineData("Cowboy", "WeeellHeeell@Yeehaw.com")]
+    [InlineData("Jackson", null)]
+    public async void CreateUser(string username, string? email)
+    {
+        // Act
+        var user = new User()
+        {
+            Name = username,
+            Email = email ?? null,
+        };
+
+        context.Users.Add(user);
+        await context.SaveChangesAsync();
+
+        // Assert
+        var DBuser = await context.Users.Where(_user => _user.Name == username).FirstOrDefaultAsync();
+        Assert.NotNull(DBuser);
+    }
+
+    [Theory]
+    [InlineData("MyEmailIsTooLong", "EEEEEEEEEEEEEEEEEEEEMMMMMMMMMMMMMMMMMMAAAAAAAAAAAIIIIIIILLLLLLL@TTTHHHIIISSSIIIISSSAAAAMMMMAAIIILLL.com")]
+    public async void CreateUserWithInvalidEmail_ThrowsException(string name, string email)
+    {
+        // Act
+        var user = new User()
+        {
+            Name = name,
+            Email = email,
+        };
+
+        context.Users.Add(user);
+        // Assert
+        await Assert.ThrowsAsync<DbUpdateException>(async() => await context.SaveChangesAsync());
+        context.Users.Remove(user);
```

Comment:
Should we maybe add a test `RemoveUser` before using it in another DbTest? :3

Reply:
Well the `context.Users.Remove()` is because otherwise other tests will try to commit that change which will then cause an exception

Reply:
I also meant: Should we perhaps Assert whether or not `context.Users.Remove(user)` has removed the user by asserting that the user does not exist after the call? :3

Reply:
Well that is kinda outside the scope of this test. `context.Users.Remove(user)` is more just cleanup so that the test doesn't interfere with other tests